xds/internal/client/client_xds.go

```diff
@@ -193,6 +189,10 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger)
 			continue
 		}
 
+		if caseSensitive := match.GetCaseSensitive(); caseSensitive != nil {
```

Comment:
Based on the comment on the proto field, if the field is not set, we should assume a value of `true`, right?
``` 
 // Indicates that prefix/path matching should be case sensitive. The default
 // is true.
```

Reply:
If we really want to be consistent, the CaseSensitive field has to be a boolean pointer. I really don't like using pointers when we can have a default value that just works, because you will need to check nil before reading its value.

Reply:
I'm not really suggesting that we use a `*bool`. But, given that when we process the received route configuration, we are anyways looking to see whether this field is set or not? So, we could still use a `caseSensitive bool` and set it up as we want based on whether this field exists, and if it exists, based on its value.

But if you are really happy with the way things are, then that's fine.

Reply:
I'm keeping it for now. If we later find this is causing problems, we can update. This is not a public API.